ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -854,84 +949,73 @@ namespace OrdinalAxis {
                     }
 
-                // Loop down along the ordinal positions. When the linear
-                // equivalent of i matches an ordinal position, interpolate
-                // between the left and right values.
-                } else {
-                    while (i--) {
-                        linearEquivalentLeft =
-                            ((ordinalSlope as any) * i) + (ordinalOffset as any);
-                        if (val >= linearEquivalentLeft) {
-                            linearEquivalentRight =
-                                ((ordinalSlope as any) *
-                                (i + 1)) +
-                                (ordinalOffset as any);
-                            // something between 0 and 1
-                            distance = (val - linearEquivalentLeft) /
-                                (linearEquivalentRight - linearEquivalentLeft);
-                            break;
-                        }
+                    const EOPlength = extendedOrdinalPositions.length,
+                        leftVisiblePoint = extendedOrdinalPositions[0],
+                        rightVisiblePoint = extendedOrdinalPositions[EOPlength - 1],
+                        slope = (rightVisiblePoint - leftVisiblePoint) / (EOPlength - 1);
+
+                    if (indexInEOP < 0) {
+                        return leftVisiblePoint + slope * indexInEOP;
                     }
+
+                    return rightVisiblePoint + slope * (indexInEOP - EOPlength);
                 }
+            } else {
+                // Even if the isInside property passed as true, due to some
+                // inaccuracy the getIndexOfPoint might return a negative value.
+                // To prevent that, compare it with zero.
+                const indexInEOP = Math.max(this.ordinal.getIndexOfPoint(val, ordinalPositions), 0),
+                    mantissa = correctFloat(indexInEOP % 1),
+                    leftNeighbour = ordinalPositions[Math.floor(indexInEOP)],
+                    rightNeighbour = ordinalPositions[Math.ceil(indexInEOP)],
+                    distance = rightNeighbour - leftNeighbour;
+
+                return ordinalPositions[Math.floor(indexInEOP)] + mantissa * distance;
 
-                // If the index is within the range of the ordinal positions,
-                // return the associated or interpolated value. If not, just
-                // return the value.
-                return (
-                    typeof distance !== 'undefined' &&
-                    typeof ordinalPositions[i] !== 'undefined' ?
-                        ordinalPositions[i] + (
-                            distance ?
-                                distance *
-                                (ordinalPositions[i + 1] - ordinalPositions[i]) :
-                                0
-                        ) :
-                        val
-                );
             }
-            return ret;
+            return val;
```

Comment:
@karolkolodziej

Reply:
I've tried to apply that simple mechanism, however, it seems like it's not that straightforward. 
It will require some additional properties, cases or if conditions that will complicate the code.  
I think that the iterative approach might be more readable in a complex calculation like this.